packages/editor/src/components/block-list/block-html.js

```diff
@@ -35,7 +35,13 @@ class BlockHTML extends Component {
 		const blockType = getBlockType( this.props.block.name );
 		const attributes = getBlockAttributes( blockType, this.state.html, this.props.block.attributes );
 		const isValid = isValidBlock( this.state.html, blockType, attributes );
+
 		this.props.onChange( this.props.clientId, attributes, this.state.html, isValid );
+
+		if ( ! isValid ) {
+			// An error was detected so flip the block back to visual mode so we see the invalid block warning
+			this.props.onToggleMode( this.props.clientId );
```

Comment:
Switching back to visual mode if the block becomes invalid may cause some troubles. Blur events are very easy to trigger, and the user may be in the invalid state temporarily while in process of writing a valid change.
I would prefer if the switch was not automatic, but I would love other options regarding the UX cc: @jasmussen, @karmatosed.
Another possibility for automatic switching to the visual mode when the block is invalid would be switching when another block is selected/block is unselected. It would also make the user notice the warning, but the switch is not as unexpected as with the blur event.

Reply:
I probably didn't explain this well in the description but nothing really changes for the user, and the behaviour is the same as currently.

When the `onBlur` fires the HTML validation is performed and the block is set as invalid, showing the invalid block message. At this point the block is still in HTML mode, although you don't see this, and the dropdown menu option will be 'Edit visually' (which we don't want).

The extra `onToggleMode` switches the block to visual mode so the menu will show `Edit as HTML'. The only change is for the menu toggle - nothing will visually change for the user as the block is invalid and will continue to show the invalid block message (i.e. the visual mode is not shown)